hivemind_core/database.py

```
@@ -1,254 +1,716 @@
+        """
+        Search for clients by a specific key-value pair in Redis.
+
+        Args:
+            key: The key to search by.
+            val: The value to search for.
+
+        Returns:
+            A list of clients that match the search criteria.
+        """
+        res = []
+        for client_id in self.redis.scan_iter(f"client:*"):
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)